distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -487,49 +486,47 @@ type Table
         this.take_end (this.length - 1) . rename_columns (Column_Mapping.By_Position new_names) on_problems=on_problems
 
 
-    ## Prototype Group By function
-    aggregate : [Aggregate_Column] -> Problem_Behavior -> Table
-    aggregate columns (on_problems=Report_Warning) =
-        validated = Aggregate_Column_Helper.prepare_aggregate_columns columns this
+    ## ALIAS group, summarize
 
-        make_key = if (validated.key_columns.length == 0) then _->(Group_By_Key.key [1]) else
-            i->(Group_By_Key.key (validated.key_columns.map v->(v.at i)))
+       Aggregates the rows in a table using any `Group_By` entries in columns.
+       The columns argument specifies which additional aggregations to perform and to return.
 
-        new_table = validated.valid_columns.map c->[c.first, Vector.new_builder]
-        aggregators = validated.valid_columns.map c->(Aggregate_Column_Aggregator.new c.second)
-        add_row _ =
-            idx = new_table.at 0 . at 1 . length
-            0.up_to (aggregators.length) . each i->
-                new_table.at i . at 1 . append ((aggregators.at i).initial)
-            idx
+       Arguments:
+       - columns: Vector of `Aggregate_Column` specifying the aggregated table.
+       - on_problems: Specifies how to handle problems if they occur, reporting
+         them as warnings by default.
 
-        # Fold
-        aggregate map i =
-            key = make_key i
-            row_index = map.get_or_else key (add_row Nothing)
+         The following problems can occur:
+         - If a column name is not in the input table, a `Missing_Input_Columns`.
+         - If a column index is out of range, a `Column_Indexes_Out_Of_Range`.
+         - If there are no valid columns in the output table, a `No_Output_Columns`.
+         - If there are invalid column names in the output table, a `Invalid_Output_Column_Names`.
+         - If there are duplicate column names in the output table, a `Duplicate_Output_Column_Names`.
+         - If grouping on or computing the `Mode` on a floating point number, a `Floating_Point_Grouping`.
+         - If an aggregation fails, an `Invalid_Aggregation_Method`.
+         - If when concatenating values there is an quoted delimited, an `Unquoted_Delimiter`
+         - If there are more than 10 issues with a single column, an `Additional_Warnings`.
```

Comment:
If each column can issue this thing, then I think it should contain the origin column as payload.

But I still think we should aggregate these issues in the same way as we do for inputs.